doc/design/adding-priority-classes.md

```diff
@@ -0,0 +1,47 @@
+# Adding Priority Classes
+
+## Description
+
+OLM supports users including `PriorityClass` objects in their bundle alongside their operator manifests. `PriorityClass`
+is used to establish a priority, or weight, to a collection of pods in order to aid the kube-scheduler when assigning pods
+to nodes. For more info, see the docs at https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/#priorityclass. 
+
+## Caveats
+
+`PriorityClass` objects are clusterwide in scope, meaning they can affect the scheduling of pods in all namespaces. All pods
+have a default priority of zero, and only those pods explicitly selected by the `PriorityClass` object will be given a priority when created.
+Existing pods running on the cluster are not affected by a new `PriorityClass`, but since clusters are dynamic and pods can be 
+rescheduled as nodes cycle in and out, a `PriorityClass` can have an impact on the long term behavior of the cluster. 
+
+Exactly one `PriorityClass` object is allowed to have the `globalDefault` setting set to true - this means that all pods on the cluster
+without an explicit priority class will use this default `PriorityClass`. Make sure you understand the effects of setting a global
+priority for pods running in the cluster before setting a `PriorityClass` as the `globalDefault`.
+
+Pods with higher priorities can preempt pods with lower priorities when they are being scheduled onto nodes: preemption can result in lower-priority pods being evicted to make room for the higher priority pod. 
+If the `PriorityClass` of the pod is extremely high (higher than the priority of core components) scheduling the pod can potentially disrupt core components running in the cluster. 
```

Comment:
When the feature is added, could we make sure that a "How to set priorities" doc exists and describes priority ranges of core kube components?

Reply:
Yes, I agree we need better docs on what objects are supported in bundles. Currently I'm not aware of any such docs and we now have multiple different sources for our docs so having an agreement on where these docs should live would be helpful.

Reply:
That being said, if we deliver the "put arbitrary kube manifests in the bundle" feature in 4.6 then we won't really need said docs - we will simply say we support all objects. 

I also don't think we need to provide a "how to set priorities doc" since its not really in the scope of OLM. It's heavily dependent on the application needs.

Reply:
We should, yes. I'm not aware of such a document existing, but it should probably be featured prominently on our new doc site